frontend/src/PlayerEditForm.tsx

```diff
@@ -0,0 +1,146 @@
+import axios from "axios";
+import React, { useEffect } from "react";
+import Box from "@mui/joy/Box";
+import Button from "@mui/joy/Button";
+import CircularProgress from "@mui/joy/CircularProgress";
+import FormControl from "@mui/joy/FormControl";
+import FormHelperText from "@mui/joy/FormHelperText";
+import FormLabel from "@mui/joy/FormLabel";
+import Sheet from "@mui/joy/Sheet";
+import Typography from "@mui/joy/Typography";
+import {
+    PlayerEditFormData,
+    ServerErrorBody,
+    ValidPlayerEditFormData,
+} from "./types";
+import TextInput from "./TextInput";
+import useFormData from "./hooks/useFormData";
+
+interface Props {
+    pid: number;
+    name: string;
+    refresh: () => void;
+}
+
+export default function PlayerEditForm({ pid, name, refresh }: Props) {
+    const initialFormData: PlayerEditFormData = {
+        pid: {
+            value: pid,
+            error: null,
+            validate: () => null,
+        },
+        newName: {
+            value: null,
+            error: null,
+            validate: function _() {
+                return this.value?.toLowerCase().trim() ===
+                    name.toLowerCase().trim()
+                    ? "Cannot rename a player to the same name"
```

Comment:
How are we not using "Fool of a Took!!" more in these strings??

Reply:
That's such a good idea 😂 we should have an issue for adding LOTR references everywhere